### ISSUE TITLE
Problem showing help when function is decorated

### ISSUE THREAD

**Issue by bzamecnik** [google/python-fire#190]
```python
import time
import fire

def decorator(f):
    def wrapper(*args, **kwds):
        return f(*args, **kwds)
    return wrapper

@decorator
def foo(name='World'):
    """
    Greets name.
    """
    print('Hello %s' % name)

if __name__ == '__main__':
    fire.Fire(foo)
```

produces:

```
# -h or --help
$ python bar.py -h
Traceback (most recent call last):
  File "bar.py", line 17, in <module>
    fire.Fire(foo)
  File "/Users/bzamecnik/.virtualenvs/rossum3/lib/python3.7/site-packages/fire/core.py", line 138, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/Users/bzamecnik/.virtualenvs/rossum3/lib/python3.7/site-packages/fire/core.py", line 471, in _Fire
    target=component.__name__)
  File "/Users/bzamecnik/.virtualenvs/rossum3/lib/python3.7/site-packages/fire/core.py", line 675, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "bar.py", line 6, in wrapper
    return f(*args, **kwds)
TypeError: foo() got an unexpected keyword argument 'h'
```

The same problem when the decorator uses `wraps()`.

```python
from functools import wraps

def timing(f):
    @wraps(f)
    def wrapper(*args, **kwds):
        return f(*args, **kwds)
    return wrapper
```

However, `python bar.py -- -h` shows the help but without the arguments:

```
NAME
    bar.py - Greets name.

SYNOPSIS
    bar.py <flags> [ARGS]...

DESCRIPTION
    Greets name.

POSITIONAL ARGUMENTS
    ARGS

FLAGS
    Flags are accepted.
```

When we remove the `@decorator` it works as expected:

```
NAME
    bar.py - Greets name.

SYNOPSIS
    bar.py <flags>

DESCRIPTION
    Greets name.

FLAGS
    --name=NAME
```

Versions: `fire=0.2.1, python 3.7.3`

## Workaround

A work around can be to move the actual code to a nested function that picks the arguments from the parent function. Then `fire` has not problem finding the arguments.

```python
from functools import wraps
import time
import fire

def timing(f):
    @wraps(f)
    def wrapper(*args, **kwds):
        return f(*args, **kwds)
    return wrapper

def foo(name='World'):
    """
    Greets name.
    """
    @timing
    def inner():
        print('Hello %s' % name)

    inner()

if __name__ == '__main__':
    fire.Fire(foo)
```

**Comment by dbieber** [comment]
I think we can fix this at least for Python 3 by switching to inspect.signature for getting the argument specs as in https://github.com/google/python-fire/issues/159.

**Comment by dbieber** [comment]
The change will go in https://github.com/google/python-fire/blob/master/fire/inspectutils.py where we currently use inspect.getfullargspec for Python 3.

**Comment by cmin764** [comment]
I can reproduce the issue too. Any known ETA for this fix?

**Comment by dbieber** [comment]
No ETA at this time, I'll update here once there's an ETA or a fix.

**Comment by dbieber** [comment]
I think we'll have the fix in soon from PR #203 and then this will go out with the next release, which  should be by end of March (no date picked out yet though).

**Comment by dbieber** [comment]
Fixed in #203 by deb33cc34f94cad86a875a9993864ac0acf93828
